File: modules/react-native-swift-ts-player/ios/SwiftTSPlayerViewManager.m
#import <React/RCTViewManager.h>

@interface RCT_EXTERN_MODULE(SwiftTSPlayerViewManager, RCTViewManager)

RCT_EXPORT_VIEW_PROPERTY(streamUrl, NSString)
RCT_EXPORT_VIEW_PROPERTY(paused, BOOL)

RCT_EXPORT_VIEW_PROPERTY(onSwiftVideoLoad, RCTDirectEventBlock)
RCT_EXPORT_VIEW_PROPERTY(onSwiftVideoError, RCTDirectEventBlock)

@end

File: modules/react-native-swift-ts-player/ios/SwiftTSPlayerViewManager.swift
import Foundation
import React

@objc(SwiftTSPlayerViewManager)
class SwiftTSPlayerViewManager: RCTViewManager {

    override func view() -> UIView! {
        return SwiftTSPlayerView()
    }

    override static func requiresMainQueueSetup() -> Bool {
        return true
    }
}
